### PR TITLE
Test the format negotiator and reorganize it

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR add a test for the ``FormatNegotiator``. Several mistakes are fixed.

### CODE REVIEW

File: Negotiation/FormatNegotiator.php [hunk 1]
```diff
@@ -73,31 +72,27 @@ public function getBest($header, array $priorities = [])
...
+                        $header .= ',';
                     }
-                }
-                if ($extensionHeader) {
                     $header .= $extensionHeader.'; q='.$options['prefer_extension'];
                 }
             }
 
-            $mimeTypes = $this->normalizePriorities(empty($priorities) ? $options['priorities'] : $priorities);
+            $mimeTypes = empty($priorities) ? $options['priorities'] : $priorities;
```
Comment (lsmith77) [inline Negotiation/FormatNegotiator.php]: does this mean that priorities now need to be mime types and we no longer support formats?

Comment (Ener-Getick) [inline Negotiation/FormatNegotiator.php]: Yes, for the moment I just removed it. I have to think how to replace this ...

Comment (lsmith77) [inline Negotiation/FormatNegotiator.php]: ok .. we should then create a follow up ticket for this. its annoying that the Request class does not give access to the mappings in ``Request::formats``.

/cc @Tobion

Comment (Ener-Getick) [inline Negotiation/FormatNegotiator.php]: We should maybe create an issue / a pr for symfony 2.8 or 3.0 to add a new method ``Request::getMimeTypes()`` and see what the symfony staff think about this.

Comment (Ener-Getick) [inline Negotiation/FormatNegotiator.php]: I created https://github.com/symfony/symfony/pull/17318

Comment (Ener-Getick) [inline Negotiation/FormatNegotiator.php]: https://github.com/symfony/symfony/pull/17113 has been merged so we need a new pr on FOSRestBundle :-)

**Pull request merged**